src/config/mongoose_config_validator_toml.erl

```diff
@@ -294,6 +295,126 @@ validate([<<"max_retry_delay">>, <<"s2s">>|Path],
+
+validate([item, <<"urls">>, item, <<"server_info">>, <<"mod_disco">>, <<"modules">>],
+         [Url]) ->
+    validate_url(Url);
+validate([item, <<"module">>, item, <<"server_info">>, <<"mod_disco">>, <<"modules">>],
+         [Mod]) ->
+    validate_module(Mod);
+validate([<<"module">>, item, <<"server_info">>, <<"mod_disco">>, <<"modules">>],
+         [all]) ->
+    ok; %% ensure that section is all or a list
+validate([<<"module">>, item, <<"server_info">>, <<"mod_disco">>, <<"modules">>],
+         [Mods]) ->
+    true = is_list(Mods); %% ensure that section is all or a list
+validate([<<"urls">>, item, <<"server_info">>, <<"mod_disco">>, <<"modules">>],
+         [Mods]) ->
+    true = is_list(Mods);
+validate([<<"name">>, item, <<"server_info">>, <<"mod_disco">>, <<"modules">>],
+         [V]) ->
+    validate_non_empty_binary(V);
+
+validate([<<"backend">>, <<"mod_inbox">>, <<"modules">>],
+         [{backend, Value}]) ->
+    validate_backend(mod_inbox, Value);
+validate([<<"aff_changes">>, <<"mod_inbox">>, <<"modules">>],
+         [{aff_changes, Types}]) ->
+    validate_boolean(Types);
+validate([<<"remove_on_kicked">>, <<"mod_inbox">>, <<"modules">>],
+         [{remove_on_kicked, Types}]) ->
+    validate_boolean(Types);
+%% Sublists for mod_inbox - ensure that they are lists
+validate([<<"reset_markers">>, <<"mod_inbox">>, <<"modules">>],
+         [{reset_markers, Types}]) ->
+    validate_list(Types);
+validate([<<"groupchat">>, <<"mod_inbox">>, <<"modules">>],
+         [{groupchat, Types}]) ->
+    validate_list(Types);
+%% Sublists for mod_inbox - items
+validate([item, <<"reset_markers">>, <<"mod_inbox">>, <<"modules">>],
+         [Type]) ->
+    validate_marker_type(Type);
+validate([item, <<"groupchat">>, <<"mod_inbox">>, <<"modules">>],
+         [Type]) ->
+    validate_groupchat_type(Type);
+
+%% One call for each rule in ip_access
+validate([_, <<"ip_access">>, <<"mod_register">>, <<"modules">>],
+         [{Policy, _Addr}]) ->
+    validate_enum(Policy, [allow, deny]);
+%% We do validations for suboptions of this option in other parts of the code.
+validate([<<"ip_access">>, <<"mod_register">>, <<"modules">>], _Value) ->
+    ok;
+validate([<<"welcome_message">>, <<"mod_register">>, <<"modules">>],
+         [{Subject, Body}]) ->
+    %% TODO We don't have any big tests for this option.
+    %% So, I don't know how it behaves with unicode characters.
+    ok;
+validate([<<"access">>, <<"mod_register">>, <<"modules">>],
+         [{access, Value}]) ->
+    validate_non_empty_atom(Value);
+validate([<<"registration_watchers">>,<<"mod_register">>,<<"modules">>],
+         [{registration_watchers, Value}]) ->
+    validate_list_of_jids(Value);
+validate([<<"password_strength">>,<<"mod_register">>,<<"modules">>],
+         [{password_strength, Value}]) ->
+    validate_non_negative_integer(Value);
+
+validate(Option, Value) ->
+    ?LOG_ERROR(#{ what => validate_unknown, option => Option, value => Value}),
```

Comment:
Currently validation is supplementary, so this is not an error.

Reply:
that line for me, ignore